packages/wallet/WalletManager.js

```diff
@@ -0,0 +1,421 @@
+    }
+    return this.wallet;
+  }
+
+  /**
+   * @returns {Object} the current wallet
+   */
+  getWallet() {
+    if (this.wallet == null) {
+      console.log('The wallet is not set.');
+      return null;
+    }
+    return this.wallet;
+  }
+
+  /**
+   * Before generating accounts, the seed and wallet MUST be there.
+   * i.e., random account generatation is NOT allowed.
+   * If the wallet is set, generate the first (0) account in the wallet;
+   * If the parameter is a number,
+   * Generate the account in the wallet. Its index is the number.
+   * If the parameter is a string,
+   * Use it as the private/public key to generate the account
+   * @returns {Account} the account created/imported
+   */
+  async getAccount() {
+    if (arguments.length === 0) {
+      if (this.wallet == null) {
+        console.log('The wallet is not initialized.');
+        return null;
+      }
+      // generate the 0th account in the wallet
+      switch (this.network) {
+        case 'stellar': {
+          this.account = new Account(this.network);
+          await this.account.setAccount(this.wallet.getKeypair(0));
+          this.accountArray.push(Object.assign({}, this.account));
+          return this.account;
+        }
+        case 'ethereum': {
+          const privateKey = `0x${this.wallet.deriveChild(0).getWallet()._privKey.toString('hex')}`;
+          this.account = new Account(this.network);
+          await this.account.setAccount(web3.eth.accounts.privateKeyToAccount(privateKey));
+          this.accountArray.push(Object.assign({}, this.account));
+          return this.account;
+        }
+        default:
+          console.log('The network is not set.');
+          return null;
+      }
+    }
+    if (arguments.length === 1 && Number.isInteger(arguments[0]) && arguments[0] >= 0) {
+      // generate the account of the wallet at the specified index
+      switch (this.network) {
+        case 'stellar': {
+          this.account = new Account(this.network);
+          await this.account.setAccount(this.wallet.getKeypair(arguments[0]));
+          this.accountArray.push(Object.assign({}, this.account));
+          return this.account;
+        }
+        case 'ethereum': {
+          const privateKey = `0x${this.wallet.deriveChild(arguments[0]).getWallet()._privKey.toString('hex')}`;
+          this.account = new Account(this.network);
+          await this.account.setAccount(web3.eth.accounts.privateKeyToAccount(privateKey));
+          this.accountArray.push(Object.assign({}, this.account));
+          return this.account;
+        }
+        default:
+          this.account = null;
+          console.log('The network has not been set.');
+      }
+      return this.account;
+    }
+    if (arguments.length === 1 && typeof (arguments[0]) === 'string') {
+      try {
+        switch (this.network) {
+          case 'stellar': {
+            if (arguments[0].charAt(0) === 'S') {
```

Comment:
Good to comment that the encoding standard is here :) https://github.com/stellar/js-stellar-base/blob/master/src/strkey.js#L8

Reply:
link added